x-pack/test/observability_ai_assistant_functional/tests/conversations/sharing.spec.ts

```diff
@@ -77,9 +79,16 @@ export default function ApiTest({ getService, getPageObjects }: FtrProviderConte
 
       describe('when changing access to Shared', () => {
         before(async () => {
-          await testSubjects.click(ui.pages.conversations.access.sharedOption);
-          await testSubjects.existOrFail(ui.pages.conversations.access.loadingBadge);
-          await testSubjects.missingOrFail(ui.pages.conversations.access.loadingBadge);
+          await interceptRequest(
+            driver.driver,
+            '*observability_ai_assistant\\/conversation\\/*',
+            (responseFactory) => {
+              return responseFactory.continue();
+            },
+            async () => {
+              await testSubjects.click(ui.pages.conversations.access.sharedOption);
+            }
+          );
```

Comment:
What's the difference between using the callback, and the flattened approach below?

```suggestion
          await interceptRequest(
            driver.driver,
            '*observability_ai_assistant\\/conversation\\/*',
            (responseFactory) => {
              return responseFactory.continue();
            }
          );
          await testSubjects.click(ui.pages.conversations.access.sharedOption);          
```

(btw. untested - I just assume this works)

Reply:
If we can avoid the callback and flatten it, we could perhaps extract it to a reusable function like:

```suggestion
          await interceptShareConversationRequest(driver);
          await testSubjects.click(ui.pages.conversations.access.sharedOption);          
```

That makes it more readable

Reply:
Ah... I just realize why it's in a callback. nevermind.